packages/lib-hedera/index.ts

```diff
@@ -0,0 +1,256 @@
+    private readonly accountId: AccountId
+  ) {
+    this.dfnsClient = dfnsOptions.dfnsClient
+    this.walletId = dfnsOptions.walletId
+  }
+
+  public static async init(options: DfnsWalletOptions): Promise<DfnsWallet> {
+    const { walletId, dfnsClient } = options
+    const res = await dfnsClient.wallets.getWallet({ walletId })
+
+    if (res.status !== 'Active') {
+      throw new DfnsError(-1, 'wallet not active', { walletId, status: res.status })
+    }
+
+    if (res.network !== 'Hedera' && res.network !== 'HederaTestnet') {
+      throw new DfnsError(-1, 'wallet is not bound to a Hedera network', {
+        walletId,
+        network: res.network,
+      })
+    }
+
+    const nativeAddressRegex = /^0\.0\.\d+$/
+    if (!nativeAddressRegex.test(res.address!)) {
+      throw new DfnsError(-1, 'account must be created. Address is not in native Hedera format (0.0.x)', {
+        walletId,
+        address: res.address!
+      })
+    }
+
+    const accountId = AccountId.fromString(res.address!)
+
+    if (!res.signingKey?.publicKey) {
+      throw new DfnsError(-1, 'wallet public key not found', { walletId })
+    }
+
+    const publicKey = PublicKey.fromString(res.signingKey.publicKey)
+
+    const client = res.network === 'HederaTestnet'
+      ? Client.forTestnet()
+      : Client.forMainnet()
+
+    return new DfnsWallet(options, new LocalProvider({ client }), publicKey, accountId)
+  }
+
+  public async signTransaction<T extends Transaction>(transaction: T): Promise<T> {
+    if (!transaction.isFrozen()) {
+      transaction.freeze()
+    }
+
+    const transactionBytes = transaction.toBytes()
+    const messageHex = '0x' + Buffer.from(transactionBytes).toString('hex')
+
+    const response = await this.dfnsClient.wallets.generateSignature({
+      walletId: this.walletId,
+      body: {
+        kind: 'Transaction',
+        transaction: messageHex,
+      },
+    })
+
+    assertSignResponseSuccessful(response)
+
+    const signature = response.signature
+    if (!signature?.encoded) {
+      throw new DfnsError(-1, 'signature encoded value missing', response)
+    }
+
+    if (!response.signedData) {
+      throw new DfnsError(-1, 'signed data missing', response)
+    }
+
+    const signedTx = Transaction.fromBytes(new Uint8Array(hexToBuffer(response.signedData)))
+
+    transaction.addSignature(this.getAccountKey(), signedTx.getSignatures())
+
+    return transaction
+  }
+
+  getProvider(): Provider {
+    return this.provider
+  }
+
+  public getLedgerId(): LedgerId | null {
+    return this.provider.getLedgerId()
+  }
+
+  public getAccountId(): AccountId {
+    return this.accountId
+  }
+
+  public getNetwork(): { [key: string]: string | AccountId } {
+    return this.provider.getNetwork()
+  }
+
+  public getMirrorNetwork(): string[] {
+    return this.provider.getMirrorNetwork()
+  }
+
+  public getAccountKey(): PublicKey {
+    return this.publicKey
+  }
+
+  public getAccountBalance(): Promise<AccountBalance> {
```

Comment:
We could have left the error ‘throw not implemented’. But since it was easy to implement... let's do it so that the user can eventually use it for complex operations.